code/__DEFINES/wiremod.dm

```diff
@@ -4,6 +4,9 @@
 /// Define to automatically handle calling the output port. Will not call the output port if the input_received proc returns TRUE.
 #define TRIGGER_CIRCUIT_COMPONENT(component, port) if(!component.input_received(port) && (component.circuit_flags & CIRCUIT_FLAG_OUTPUT_SIGNAL)) component.trigger_output.set_output(COMPONENT_SIGNAL)
 
+/// Handles adding option ports more simply
+#define SET_OPTION_LIST(option_list) port_type = /datum/port/input/option, extra_args = list(possible_options = option_list)
```

Comment:
I don't really like macros like these, can you just make creating an option its own proc so we don't do anything funky?

Reply:
sure